karyon2-admin/src/main/java/netflix/adminresources/AdminResourcesContainer.java

```diff
@@ -144,15 +141,25 @@ public void init() throws Exception {
                 server.start();
 
                 final Connector connector = server.getConnectors()[0];
-                listenPort = connector.getLocalPort();
+                serverPort = connector.getLocalPort();
             }
         } catch (Exception e) {
             logger.error("Exception in building AdminResourcesContainer ", e);
         }
     }
 
+    private void initAdminContainerConfigIfNeeded() {
+        if (adminContainerConfig == null) {
+            adminContainerConfig = new AdminConfigImpl();
+        }
+    }
+
     public int getServerPort() {
-        return listenPort;
+        return serverPort;
+    }
+
+    public AdminPageRegistry getAdminPageRegistry() {
+        return adminPageRegistry;
     }
 
     private Module getAdditionalBindings() {
```

Comment:
This method enforces AdminConfigImpl binding, while it should be possible to inject custom instance of it. Now injecting custom implementation results in duplicate binding exception.